File: src/ca/uqam/mgl7460/tp1/types/modeles/DemandeurPret.java
package ca.uqam.mgl7460.tp1.types.modeles;

/**
 * Cette classe représente les demandeur-se-s de prêt
 */
public interface DemandeurPret {

    /**
     * retourne le nom.
     * @return
     */
    public String getNom();

    /**
     * retourne le prenom
     * @return
     */
    public String getPrenom();

    /**
     * retourne le numéro d'assurance sociale
     * @return
     */
    public String getNas();

    /**
     * retourne le revenu annuel brut
     */
    public float getRevenuAnnuel();

    /**
     * modifie le revenu annuel brut
     * @param revenu
     */
    public void setRevenuAnnuel(float revenu);

    /**
     * retourne les obligations annuelles de l'emprunteur-se, 
     * c'est à dire, les choses qu'il/elle doit payer, qui sont 
     * incompressibles, telles les frais de logement, transport pour
     * se rendre au travail, etc. En d'autre termes, les dépenses
     * incompressibles.
     * @return
     */
    public float getObligationsAnnuelles();

    /**
     * cette méthode modifie les obligations annuelles de l'emprunteur-se
     * @param obligations
     */
    public void setObligationsAnnuelles(float obligations);

    /**
     * cette méthode retourne le taux d'endettement de l'emprunteur-se
     * ("debt over income ratio"), qui consiste, simplement, en le
     * ratio obligationsAnnuelles/revenuAnnuel
     * @return
     */
    public float getTauxEndettement();

    /**
     * retourne la cote de crédit de l'emprunteur-se
     * @return
     */
    public int getScoreCredit();

    /**
     * modifie la cote de crédit de l'emprunteur-se
     * @param score
     */
    public void setScoreCredit(int score);
}

File: src_python/modeles/demandeur_pret.py

class DemandeurPret:
    def __init__(self, prenom, nom, nas, revenuAnnuel= 0.0, obligationsAnnuelles=0.0, scoreCredit=0):
        self.prenom = prenom
        self.nom = nom
        self.nas = nas

        self.revenuAnnuel = revenuAnnuel
        self.obligationsAnnuelles = obligationsAnnuelles
        self.scoreCredit = scoreCredit
    
    @property
    def nom(self):
        return self.nom
    
    @property 
    def prenom(self):
        return self.prenom
    
    @property
    def nas(self):
        return self.nas
    
    @property
    def revenuAnnuel(self):
        return self._revenuAnnuel
    
    @revenuAnnuel.setter
    def revenuAnnuel(self, value):
        self._revenuAnnuel = value

    @property
    def obligationsAnnuelles(self):
        return self._obligationsAnnuelles
    
    @obligationsAnnuelles.setter
    def obligationsAnnuelles(self, value):
        self._obligationsAnnuelles = value
    
    @property
    def scoreCredit(self):
        return self._scoreCredit
    
    @scoreCredit.setter
    def scoreCredit(self, value):
        self._scoreCredit = value

    @property
    def tauxEndettement(self):
        if self.revenuAnnuel == 0:
            return 0.0
        return self.obligationsAnnuelles / self.revenuAnnuel


